### PR TITLE
Add workaround for Julia 1.12.1

### PR DESCRIPTION
**Pull request by omus**
Adding CI tests against the latest releases of Julia to ensure we've retained compatibility with Julia 1.12.0 and 1.12.1.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1211867634077782

### CODE REVIEW

**Comment by omus** [comment]
Seeing failures like this on Julia 1.12.1:
```
ERROR: LoadError: Unknown package: Preferences
```

**Comment by omus** [comment]
Looks like this was a side effect of https://github.com/JuliaLang/julia/pull/59212

**Comment by omus** [comment]
I've made an issue in the Julia repo about this: https://github.com/JuliaLang/julia/issues/60077

File: pkg-precompile.jl [hunk 1]
```diff
@@ -272,6 +272,15 @@ path_tracked_pkgs = [PkgId(uuid, dep.name) for (uuid, dep) in Pkg.dependencies(e
                      if dep.is_tracking_path]
 setdiff!(precompile_pkgs, path_tracked_pkgs)
 
+# Julia 1.12.1 no longer allows initiating precompilation for packages which are indirect
+# dependencies. It is probably reasonable to use this approach on all versions of Julia. We
+# can make that call once we get feedback on: https://github.com/JuliaLang/julia/issues/60077
+if VERSION >= v"1.12.1"
+    indirect_dep_pkgs = [PkgId(uuid, dep.name) for (uuid, dep) in Pkg.dependencies(env)
+                     if !dep.is_direct_dep]
```
Comment (omus) [inline pkg-precompile.jl]:
```suggestion
                         if !dep.is_direct_dep]
```

**Review by iamed2 — APPROVED** [review]
Do we gain anything by attempting to precompile indirect dependencies on other Julia versions? Won't those get precompiled automatically anyway when the direct dependencies are precompiled?

**Comment by omus** [comment]
> Do we gain anything by attempting to precompile indirect dependencies on other Julia versions? Won't those get precompiled automatically anyway when the direct dependencies are precompiled?

I thought about this and I think the original intent was to just be explicit about the packages we care about precompiling. Just passing in the direct dependencies should be good enough. If there is a corner case where being explicit about all dependencies is required then our tests currently don't catch that. As this is intended to just be a bugfix limiting the change to just 1.12.1+ should allow us to catch scenarios we haven't considered without breaking older Julia versions.
